src/Main.java

```diff
@@ -1,17 +1,87 @@
-// Press Shift twice to open the Search Everywhere dialog and type `show whitespaces`,
-// then press Enter. You can now see whitespace characters in your code.
+import product.Product;
+import product.ProductNotFoundException;
+import product.ProductService;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.*;
+
+
 public class Main {
     public static void main(String[] args) {
-        // Press Alt+Enter with your caret at the highlighted text to see how
-        // IntelliJ IDEA suggests fixing it.
-        System.out.printf("Hello and welcome!");
+        List<Product> products = new ArrayList<>();
+        products.add(new Product("Book", new BigDecimal("300"), false, LocalDateTime.now()));
+        products.add(new Product("Book", new BigDecimal("200"), true, LocalDateTime.now()));
+        products.add(new Product("Electronics", new BigDecimal("500"), true, LocalDateTime.now()));
+        products.add(new Product("Book", new BigDecimal("150"), true, LocalDateTime.now()));
+        products.add(new Product("Clothing", new BigDecimal("50"), true, LocalDateTime.now()));
+
+        // 1.2
+        List<Product> expensiveBooks = ProductService.getExpensiveBooks(products);
+
+        System.out.println("Expensive Books:");
+        expensiveBooks.forEach(System.out::println);
+
+        // 2.2
+        BigDecimal discountPercentage = new BigDecimal("0.10");
```

Comment:
why did you decided to use BigDecimal? it is not a mistake but float would be better